src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridExtensions.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Azure.EventGrid.Models;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+	/// <summary>
+	/// 
+	/// </summary>
+	public static class EventGridEventExtensions
+    {
+		private static string dataSyncToken = "syncToken";
+
+		/// <summary>
+		/// This method uses an EventGridEvent from EventGrid and tries to create a <see cref="PushNotification"/>
+		/// </summary>
+		/// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+		/// <param name="pushNotification"> out parameter set up in this method</param>
+		/// <returns></returns>
+		public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+		{
+			pushNotification = null;
+
+			if (eventGridEvent == null || eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
+			{
+				return false;
+			}
+
+			try
+			{
+				string eventGridData = eventGridEvent.Data.ToString();
+				JsonElement jsonMessage = JsonDocument.Parse(eventGridData).RootElement;
+				string syncToken = jsonMessage.GetProperty(dataSyncToken).GetString();
+
+				pushNotification = new PushNotification()
+				{
+					SyncToken = syncToken,
+					EventType = eventGridEvent.EventType,
+					ResourceUri = new Uri(eventGridEvent.Subject)
+				};
+				return true;
+			}
+			catch (JsonException) { }
+			catch (ArgumentNullException) { }
+			catch (DecoderFallbackException) { }
+			catch (ArgumentException) { }
+			catch (AmbiguousMatchException) { }
+			catch (NullReferenceException) { }
+			catch (InvalidOperationException) { }
+			catch (KeyNotFoundException) { }
+			catch (UriFormatException) { }
```

Comment:
Why are we catching `DecoderFallbackException`, `AmbiguousMatchException`, `NullReferenceException`?
The only possible exceptions come from the following calls:
1. eventGridEvent.Data.ToString() - [BinaryData.ToString()](https://docs.microsoft.com/en-us/dotnet/api/system.binarydata.tostring?view=dotnet-plat-ext-5.0) does not throw any exception.
2. [JsonDocument.Parse(string)](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsondocument.parse?view=net-5.0#System_Text_Json_JsonDocument_Parse_System_String_System_Text_Json_JsonDocumentOptions_) only throws `JsonException` and `ArgumentException`. We should catch both.
3. [JsonElement.GetProperty(string)](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getproperty?view=net-5.0#System_Text_Json_JsonElement_GetProperty_System_String_)  only throws `InvalidOperationException` and `KeyNotFoundException` in addition to the two in my previous point. We should catch both. No need to catch `ArgumentNullException` and `ObjectDisposedException` because we already caught their base exceptions.
4. [JsonElement.GetString()](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getstring?view=net-5.0) - no new exceptions other than the ones already discussed above.
5. [Uri(String)](https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-5.0#System_Uri__ctor_System_String_) throws `UriFormatException`

So these are the only 5 exceptions that we should catch here.

Reply:
Shouldn't we catch the most specific exception for each case? So including the `ObjectDisposedException` and `ArgumentNullException`?
The first three you mentioned I checked again and have removed them

Reply:
Yes, we should catch the most specific exception. But in this case, ObjectDisposedException and ArgumentNullException will always be caught by the catch blocks of InvalidOperationException and ArgumentException respectively. So adding (or not adding) separate catch blocks for ObjectDisposedException and ArgumentNullException will have no effect.